src/vm/invokeutil.cpp

```diff
@@ -666,6 +671,9 @@ OBJECTREF InvokeUtil::CreateObject(TypeHandle th, void * pValue) {
     MethodTable *pMT = NULL;
```

Comment:
`GetSignatureCorElementType` is a regular element type that you find in metadata signatures, etc.

`GetInternalCorElementType` used by the caller is relevant for calling convention - what gets passed in registers, etc.

The bug that broke corefx is when these two differ.

Reply:
Yeah, it broke enums. I'm thinking we should just strip `CreateObject` down to handling objects, pointers and function pointers and let the prealloc-and-copy code handle all value types.